### PR TITLE
option$buttons works for a scalar string and a boolean value

### LINKED ISSUE #658: CSV, COPY and PRINT buttons showing also when requesting only a single buttons
I want the datatable with a single button, say PDF. But when I use the extension with a single value for the buttons, other buttons get included too (CSV, COPY and PRINT).

Maybe the default options for the buttons extension get loaded when the number of buttons is one, instead of only loading them on empty / `NULL`.

The examples bellow are the same from the [documentation](https://rstudio.github.io/DT/extensions.html)

```
datatable(
  iris, extensions = 'Buttons', options = list(
    dom = 'Bfrtip',
    buttons = 'pdf'
  )
)
```
PDF button + CSV, COPY and PRINT
image

```
datatable(
  iris, extensions = 'Buttons', options = list(
    dom = 'Bfrtip',
    buttons = c('csv')
  )
)
```
3 buttons here CSV, COPY and PRINT
image

Session info

```
> xfun::session_info()
R version 3.5.3 (2019-03-11)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.2 LTS, RStudio 1.1.463

Locale:
  LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8       
  LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
  LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C              
  LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

Package version:
  assertthat_0.2.1   BH_1.69.0.1        cli_1.1.0          colorspace_1.4.1  
  compiler_3.5.3     crayon_1.3.4       crosstalk_1.0.0    digest_0.6.18     
  DT_0.5.3           fansi_0.4.0        ggplot2_3.1.1      glue_1.3.1        
  graphics_3.5.3     grDevices_3.5.3    grid_3.5.3         gtable_0.3.0      
  htmltools_0.3.6    htmlwidgets_1.3    httpuv_1.5.1       jsonlite_1.6      
  labeling_0.3       later_0.8.0        lattice_0.20.38    lazyeval_0.2.2    
  magrittr_1.5       MASS_7.3.51.1      Matrix_1.2.17      methods_3.5.3     
  mgcv_1.8.28        mime_0.6           munsell_0.5.0      nlme_3.1.139      
  pillar_1.3.1       pkgconfig_2.0.2    plyr_1.8.4         promises_1.0.1    
  R6_2.4.0           RColorBrewer_1.1.2 Rcpp_1.0.1         reshape2_1.4.3    
  rlang_0.3.4        scales_1.0.0       shiny_1.3.1        sourcetools_0.1.7 
  splines_3.5.3      stats_3.5.3        stringi_1.4.3      stringr_1.4.0     
  tibble_2.1.1       tools_3.5.3        utf8_1.1.4         utils_3.5.3       
  viridisLite_0.3.0  withr_2.1.2        xfun_0.6           xtable_1.8-3      
  yaml_2.2.0  
```

---

By filing an issue to this repo, I promise that

- [x] I have fully read the issue guide at https://yihui.name/issue/.
- [x] I have provided the necessary information about my issue.
    - If I'm asking a question, I have already asked it on Stack Overflow or RStudio Community, waited for at least 24 hours, and included a link to my question there.
    - If I'm filing a bug report, I have included a minimal, self-contained, and reproducible example, and have also included `xfun::session_info('DT')`. I have upgraded all my packages to their latest versions (e.g., R, RStudio, and R packages), and also tried the development version: `remotes::install_github('rstudio/DT')`.
    - If I have posted the same issue elsewhere, I have also mentioned it in this issue.
- [x] I have learned the Github Markdown syntax, and formatted my issue correctly.

### PR DESCRIPTION
**Pull request by shrektan**
Closes #658

Now the code below works.

```r
library(DT)
datatable(
  iris, extensions = 'Buttons', options = list(
    dom = 'Bfrtip',
    buttons = c('pdf')
  )
) 
```

### CODE REVIEW

File: R/datatables.R [hunk 1]
```diff
@@ -102,6 +102,13 @@ datatable = function(
     getOption('DT.options', list()),
     if (is.function(options)) options() else options
   )
+
+  # https://github.com/rstudio/DT/issues/658
+  # must provide a list to options$buttons.
+  # otherwise, scalar string doesn't work expectedly.
+  if (is.character(options[['buttons']]))
```
Comment (yihui) [inline R/datatables.R]: Is this test necessary? I mean, can we unconditionally convert `options[['buttons']]` to a list via `as.list()`? If it is already a list, it won't hurt.

Comment (shrektan) [reply R/datatables.R]: I agree with you.

Comment (yihui) [reply R/datatables.R]: Well, I think it is still necessary because `buttons` can take a boolean value: https://datatables.net/reference/option/buttons

Comment (shrektan) [reply R/datatables.R]: Yes, but unfortunately, DT doesn't support a boolean value at the present, due to :

https://github.com/rstudio/DT/blob/c6fd8641d551e6f130f5512f9334c8e329228c70/R/datatables.R#L575-L576

**Comment by shrektan** [comment]
Now DT supports a boolean value as well.

```r
library(DT)
# display all, pdf works fine
datatable(
  iris, extensions = 'Buttons', options = list(
    dom = 'Bfrtip',
    buttons = TRUE
  )
) 

# display none buttons
datatable(
  iris, extensions = 'Buttons', options = list(
    dom = 'Bfrtip',
    buttons = FALSE
  )
) 

# display only pdf, which works
datatable(
  iris, extensions = 'Buttons', options = list(
    dom = 'Bfrtip',
    buttons = 'pdf'
  )
) 

# display none buttons
datatable(
  iris, extensions = 'Buttons', options = list(
    dom = 'Bfrtip',
    buttons = character()
  )
) 

```

**Pull request merged**